addons/survey/models/survey_survey.py

```diff
@@ -552,11 +677,42 @@ def action_survey_preview_certification_template(self):
             'url': '/survey/%s/get_certification_preview' % (self.id)
         }
 
+    def action_start_session(self):
+        """ Sets the necessary fields for the session to take place and starts it. """
+        self.ensure_one()
+        self.write({
+            'questions_layout': 'page_per_question',
+            'session_current_question_id': None,
+            'session_state': 'ready'
+        })
+        return self.action_open_session_manager()
+
+    def action_open_session_manager(self):
+        self.ensure_one()
+
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Open Session Manager",
+            'target': 'self',
+            'url': '/survey/session_manage/%s' % self.access_token
+        }
+
+    def action_end_session(self):
+        self.ensure_one()
+
+        self.write({'session_state': 'not_started'})
+        self.user_input_ids.sudo().write({'state': 'done'})
+        self.env['bus.bus'].sendone(self.access_token, {'type': 'end_session'})
+
     def get_start_url(self):
-        return 'survey/start/%s' % self.access_token
+        return '/survey/start/%s' % self.access_token
```

Comment:
Why this change ? Is it a fix ?

Reply:
Needed at some point but not anymore, I reverted.